src/pages/Thanks/ThanksPageOptIn.vue

```diff
@@ -0,0 +1,369 @@
+import confetti from 'canvas-confetti';
+import CheckoutReceipt from '@/components/Checkout/CheckoutReceipt';
+import SocialShareV2 from '@/components/Checkout/SocialShareV2';
+import BorrowerImage from '@/components/BorrowerProfile/BorrowerImage';
+import { formatDistanceToNowStrict } from 'date-fns';
+import GuestAccountCreation from '@/components/Forms/GuestAccountCreation';
+import KvButton from '~/@kiva/kv-components/vue/KvButton';
+import KvMaterialIcon from '~/@kiva/kv-components/vue/KvMaterialIcon';
+
+const getLoans = receipt => {
+	const loansResponse = receipt?.items?.values ?? [];
+	const loans = loansResponse
+		.filter(item => item.basketItemType === 'loan_reservation')
+		.map(item => {
+			return {
+				...item.loan,
+				team: item.team,
+			};
+		});
+
+	return loans;
+};
+
+export default {
+	name: 'ThanksPageOptIn',
+	components: {
+		WwwPage,
+		KvButton,
+		OptInSteps,
+		KvExpandable,
+		KvMaterialIcon,
+		CheckoutReceipt,
+		SocialShareV2,
+		BorrowerImage,
+		GuestAccountCreation,
+	},
+	inject: ['apollo', 'cookieStore'],
+	data() {
+		return {
+			openOrderConfirmation: false,
+			openShareModule: false,
+			openCreateAccount: false,
+			mdiChevronDown,
+			userOptedOut: false,
+			isGuest: false,
+			lender: {},
+			loans: [],
+			receipt: null,
+		};
+	},
+	apollo: {
+		preFetch(config, client, { cookieStore, route }) {
+			const transactionId = route.query?.kiva_transaction_id
+				? numeral(route.query?.kiva_transaction_id).value()
+				: null;
+
+			return client.query({
+				query: thanksPageQuery,
+				variables: {
+					checkoutId: transactionId,
+					visitorId: cookieStore.get('uiv') || null,
+				}
+			});
+		}
+	},
+	computed: {
+		selectedLoan() {
+			/**  We should select a loan if we are going to ask for comments for it.
+			* The priority order is:
+			* 1. PFP loan
+			* 2. Partner loan With Team Attribution
+			* 3. Loan with the highest unreservedAmount
+			* loans should be sorted by unreservedAmount.
+			*/
+			const orderedLoans = orderBy(this.loans, ['unreservedAmount'], ['desc']);
+			if (this.hasPfpLoan) {
+				return orderedLoans.find(loan => loan.inPfp);
+			}
+			if (this.hasTeamAttributedPartnerLoan) {
+				return orderedLoans.find(loan => loan?.team?.name);
+			}
+			return orderedLoans[0] || {};
+		},
+		hasPfpLoan() {
+			return this.loans.some(loan => loan.inPfp);
+		},
+		hasTeamAttributedPartnerLoan() {
+			return this.loans.some(loan => loan?.distributionModel === 'fieldPartner' && loan?.team?.name);
+		},
+		borrowerName() {
+			return this.selectedLoan?.name ?? '';
+		},
+		isGuestUsCheckout() {
+			// Is a guest checking out only with US loans?
+			// eslint-disable-next-line no-underscore-dangle
+			return this.isGuest && this.loans.every(loan => loan?.__typename === 'LoanDirect');
+		},
+		hash() {
+			return this.selectedLoan.image?.hash ?? '';
+		},
+		weeksToRepay() {
+			const date = Date.parse(this.selectedLoan?.disbursalDate) ?? new Date();
```

Comment:
The date that a loan will start to be repaid is different depending on the loan. For loans that have a fixed repayment plan you can get the date from the first element of the expectedPayments array:
```graphql
... on LoanBasic {
  terms {
    expectedPayments {
      dueToKivaDate
    }
  }
}
```
For loans that don't have any expected payments yet, we can assume that the first repayment will start 1 month after the loan finishes fundraising. We don't know when the loan will finish fundraising, just that it will be between now and the `plannedExpirationDate`. So it's actually a range of time that the first repayment could happen, and we need to calculate the beginning and end of that range and translate it to something like "2-3 weeks" or "3-5 weeks".